spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -459,6 +459,7 @@ protected void actualStart() throws Exception {
 								}
 								this.logger.error("Error creating consumer; retrying in " + nextBackOff, e);
 								doShutdown();
+								this.consumersToRestart.clear();
```

Comment:
OK. Thank you for your thoughts.
Well, IMO it looks more logical do not add to the `consumersToRestart` because of some condition. 
While  this `consumersToRestart.clear()` works because of the synchronization blocking in the monitor thread, that doesn't mean that it is going to work in the future because we will change the threading model somehow. So, our lucky state here might not be possible and monitor thread may simply go ahead and iterate over `consumersToRestart` in parallel with this start loop.

The `if (started)` is much robust IMO.

Does it make sense to you?